include/boost/url/grammar/detail/any_iter_rule.hpp

```diff
@@ -0,0 +1,242 @@
+//
+// Copyright (c) 2019 Vinnie Falco (vinnie.falco@gmail.com)
+//
+// Distributed under the Boost Software License, Version 1.0. (See accompanying
+// file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+//
+// Official repository: https://github.com/boostorg/url
+//
+
+#ifndef BOOST_URL_DETAIL_ANY_ITER_RULE_HPP
+#define BOOST_URL_DETAIL_ANY_ITER_RULE_HPP
+
+#include <boost/static_assert.hpp>
+#include <atomic>
+#include <cstdlib>
+#include <type_traits>
+
+namespace boost {
+namespace urls {
+namespace grammar {
+namespace detail {
+
+template<class T>
+struct BOOST_SYMBOL_VISIBLE
+    any_iter_rule
```

Comment:
To be honest I usually miss some small comments to avoid spending time reverse-engineering other classes to completely understand this one.